reference/src/taimpl/src/internal/ta.c

```diff
@@ -426,74 +428,104 @@ static sa_status ta_invoke_key_provision(
-	       in_wrapping_key = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKey;
-	       in_wrapping_key_length = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKeyLength;
-
-	       /*TODO SoC Vendor:here just provide an example to pass ESN data,
+               in_encryption_key = ((NetflixProvisioning*)params[1].mem_ref)->encryptionKey;
+               in_encryption_key_length = ((NetflixProvisioning*)params[1].mem_ref)->encryptionKeyLength;
+               in_hmac = ((NetflixProvisioning*)params[1].mem_ref)->hmacKey;
+               in_hmac_length = ((NetflixProvisioning*)params[1].mem_ref)->hmacKeyLength;
+               in_wrapping_key = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKey;
+               in_wrapping_key_length = ((NetflixProvisioning*)params[1].mem_ref)->wrappingKeyLength;
+               INFO("in_encryption_key:0x%x, in_encryption_key_length:%d",
+                   in_encryption_key, in_encryption_key_length);
+               INFO("in_hmac:0x%x, in_hmac_length:%d",
+                   in_hmac, in_hmac_length);
+               INFO("in_wrapping_key:0x%x, in_wrapping_key_length:%d",
+                   in_wrapping_key, in_wrapping_key_length);
+
+               /*TODO SoC Vendor:here just provide an example to pass ESN data,
                  how to use it is all up to SOC vendor.
                 */
-              {
-	         void *esn = ((NetflixProvisioning*)params[1].mem_ref)->esnContainer;
-	         size_t esnLength = ((NetflixProvisioning*)params[1].mem_ref)->esnContainerLength;
-	         INFO("esn:0x%x, esnLength:%d", esn, esnLength);
-	      }
-              break;
+               {
+                    void *esn = ((NetflixProvisioning*)params[1].mem_ref)->esnContainer;
+                    size_t esnLength = ((NetflixProvisioning*)params[1].mem_ref)->esnContainerLength;
+                    INFO("esn:0x%x, esnLength:%d", esn, esnLength);
+               }
+               break;
 	  default:
-	      ERROR("unknown key provisioning type");
-              break;
+	       ERROR("unknown key provisioning type");
+               break;
      }
      INFO("ta_key_type: %d", ta_key_type);
      INFO("in_length: %d", in_length);
      INFO("key_import->key: 0x%x", key_provision_ta->key);
      sa_status status = SA_STATUS_OK;
      if (WIDEVINE_OEM_PROVISIONING == ta_key_type ||
-	 PLAYREADY_MODEL_PROVISIONING == ta_key_type) {	
+         PLAYREADY_MODEL_PROVISIONING == ta_key_type) {
          status = ta_sa_key_import(&key_provision_ta->key, key_provision_ta->key_format,
-	    in,in_length,parameters, context->client, uuid);
+             in,in_length,parameters, context->client, uuid);
+         if (SA_STATUS_OK != status) {
+	     ERROR("ta_sa_key_import failed");
+	     return status;
+	 }
      } else if (NETFLIX_PROVISIONING == ta_key_type) {
          status = ta_sa_key_import(&key_provision_ta->key, key_provision_ta->key_format,
```

Comment:
I don't think this is correct. We shouldn't be using the ta_sa_key_import function. We created ta_sa_key_provision to handle all the TA provisioning details. For Netflix, ta_sa_key_provision should take the entire Netflix provisioning bundle rather than having to make 3 separate calls, one for each key type.